http/README.md

```diff
@@ -1,172 +1,106 @@
-
-## Mapping to libp2p Streams
-
-libp2p services whose service is specified as request-response protocols can use a single protocol implementation to make the service available over HTTP as well as on top of libp2p streams.
-
-The libp2p protocol identifier is `/http1.1`. After negotiating this protocol using multistream-select, nodes treat the stream as a HTTP/1.1 stream for a single HTTP request (i.e. nodes MUST NOT use request pipelining).
+The resource contains a mapping of application protocols to their respective URL. For example, this configuration file would tell a client
 
-## Outlook: Interaction with Intermediaries
+1. That the Kademlia protocol is available at `/kademlia` and
+2. The [IPFS Path Gateway API](https://specs.ipfs.tech/http-gateways/path-gateway/) is mounted at `/`.
 
-One of the advantages of running HTTP is that there’s widely deployed caching infrastructure (CDNs). Content-addressed data is infinitely cacheable. Assuming a properly design data transfer protocol, retrieval for CIDs could be cached by the CDN and made available via a POP (geographically) close to the user, dramatically reducing retrieval latencies.
+It is valid expose a service at `/`. It is RECOMMENDED that the server resolve more specific URLs before less specific ones. e.g. a path of `/kademlia/foo` should be routed to the Kademlia protocol rather than the IPFS HTTP API.
 
-Services SHOULD specify the caching properties (if any), and set the appropriate cache headers (according to RFC 9111).
+## Peer ID Authentication
 
-CDNs can also be used to increase censorship resistance, since the CDN effectively hides the IP address of the origin server. With the upcoming introduction of ECHO (Encrypted ClientHello) in TLS, all that an on-path observer will be able to see is that a client is establishing a connection to a certain CDN, but not to which domain name.
+When using the HTTP Transport, peer id authentication is optional. You only pay for it if you need it. This is benefits use cases that don’t need peer authentication (e.g. fetching content addressed data) or authenticate some other way (not tied to libp2p peer ids).
 
-The level of delegation between the origin node and the CDN can be adjusted. In the simplest configuration, the origin node is the only node that holds the libp2p private key, thus requests to the `server-auth` protocol would be forwarded from the CDN to the origin server. In a more advanced configuration, it would be possible to move the private key to a worker on the edge of the CDN, and perform the signing operation there (thereby reducing the request latency for `server-auth` requests).
+Peer ID authentication in the HTTP Transport follows a similar to pattern to how libp2p adds Peer ID authentication in WebTransport and WebRTC. We run the standard libp2p Noise handshake, but using `IX` for client and server authentication or `NX` for just server authentication.
 
-## FAQ
+### Authentication flow
```

Comment:
This seems fine for now, but should we point out that this is just _one_ method you can do authentication? It would be nice to use https://datatracker.ietf.org/doc/draft-ietf-httpbis-unprompted-auth/ once it becomes available.

Reply:
Agreed that this is only one way you can do _peer id_ authentication. I'll mention that, thanks.

Reply:
The preceding section covers this already:

> This benefits use cases that don’t need peer authentication (e.g. fetching content addressed data) or *authenticate some other way (not tied to libp2p peer ids).*

Any suggestions to make this clearer?

Reply:
It's _one way_ to do **libp2p peer authentication**. We might come up with better ways later (e.g. the draft I linked above).

Reply:
Something to that affect is on the other pr